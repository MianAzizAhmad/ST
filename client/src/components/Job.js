import moment from 'moment'
import { Link } from 'react-router-dom'
import Wrapper from '../assets/wrappers/Job'
import react, {useState} from "react";

const Job = ({
  _id,
  position,
  company,
  jobLocation,
  job,
  createdAt,
  status,
}) => {
  const [jobType, setJobType] = useState({Type : job});

  const  setEditJob = () => {
    setJobType({Type : "Clerk"});
  };

  let date = moment(createdAt)
  date = date.format('MMM Do, YYYY')
  return (
    <Wrapper>
      <header>
        <div className='main-icon'>{company}</div>
        <div className='info'>
          <h5>{position}</h5>
          <p>{company}</p>
        </div>
      </header>
      <div className='content'>
        <div className='content-center'>
          <span className='icon'>{date}</span>
          <span className='text' id="type">{jobType.Type}</span>
          <div className={`status ${status}`}>{status}</div>
        </div>
        <footer>
          <div className='actions'>
            <button
              type='button'
              className='btn edit-btn'
              id="edit"
              onClick={() => setEditJob()}
            >
              Edit
            </button>
          </div>
        </footer>
      </div>
    </Wrapper>
  )
}

export default Job
