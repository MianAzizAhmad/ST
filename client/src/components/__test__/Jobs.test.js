import React from 'react';
import Job from './../Job';
import "@testing-library/jest-dom/extend-expect"
import { fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Jobs Test Case { ', ()=>{

    const fn = jest.fn();

    it(' Snap 1 }', ()=>{
        const component = renderer.create(<Job _id="1"
        position="Fresh" 
        company="Confiz"
        jobLocation="Gulberg"
        job="Associate Developer"
        createdAt="HR"
        status="Offered"></Job>).toJSON();
        expect(component).toMatchSnapshot();
    });

    it(' Snap 2 }', ()=>{
      
        const component = renderer.create(<Job _id="1"
        position="Fresh" 
        company="Confiz"
        jobLocation="Gulberg"
        job="Associate Developer"
        createdAt="HR"
        status="Offered" onClick={fn}></Job>).toJSON();
        const button = tree.root.getElementbyId('edit');
        button.props.onClick()
        // Verify callback is invoked
        expect(fn.mock.calls.length).toBe(1);
        expect(tree.root.getElementbyId('type').value).toBe("Clerk");
    });
});