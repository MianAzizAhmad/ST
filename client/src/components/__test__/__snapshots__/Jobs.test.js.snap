// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs Test Case {   Snap 1 } 1`] = `
<article
  className="sc-bczRLJ hMCfpN"
>
  <header>
    <div
      className="main-icon"
    >
      Confiz
    </div>
    <div
      className="info"
    >
      <h5>
        Fresh
      </h5>
      <p>
        Confiz
      </p>
    </div>
  </header>
  <div
    className="content"
  >
    <div
      className="content-center"
    >
      <span
        className="icon"
      >
        Invalid date
      </span>
      <span
        className="text"
        id="type"
        title="type"
      >
        Associate Developer
      </span>
      <div
        className="status Offered"
      >
        Offered
      </div>
    </div>
    <footer>
      <div
        className="actions"
      >
        <button
          className="btn edit-btn"
          id="edit"
          onClick={[Function]}
          type="button"
        >
          Edit
        </button>
      </div>
    </footer>
  </div>
</article>
`;
